Updating url to indicate if url is present or absent in config and cluster respectively

mohammedzee1000 opened the pull request:
## What is the purpose of this change? What does it change?
<!-- Describe the changes here, as detailed as needed. -->
 - URL list now also sees urls in cluster but not in config
 - URLs presence or absence of config is shown
 - The push command in URL in unified and in case of list, now only appears when there is url changes to push.

## Was the change discussed in an issue?
fixes  #1650
unification of messaging still needs to happen though #1737
<!-- Please do Link issues here. -->

## How to test changes?
<!-- Please describe the steps to test the PR -->
```
$ odo component list
ACTIVE     NAME            TYPE
*          nodejs-qdka     nodejs

$ odo url list      
 ✗  no URLs found for component nodejs-qdka in application app

$ odo url create --port 8080
 ✓  URL created for component: nodejs-qdka

To create URL on the OpenShift Cluster, please use `odo push` 

$ odo url list
Found the following URLs for component nodejs-qdka in application app:
NAME                 CONFIG      URL                          PORT
nodejs-qdka-8080     Present     <not created on cluster>     8080
To create URLS on the OpenShift Cluster, please use `odo push` 

$ odo push
 ✓  Checking component
 ✓  Checking component version
 ◑  Applying component settings to component: nodejs-qdka ✓  Checking URL nodejs-qdka-8080
 ✓  Successfully created URL for component: nodejs-qdka
 ✓  http://nodejs-qdka-8080-app-myproject.192.168.42.180.nip.io
 ✓  Applying component settings to component: nodejs-qdka
 ✓  Successfully updated component with name: nodejs-qdka
 ✓  Pushing changes to component: nodejs-qdka of type local
 ✓  Waiting for component to start
 ✓  Copying files to component
 ✓  Building component
 ✓  Changes successfully pushed to component: nodejs-qdka

$ odo url list
Found the following URLs for component nodejs-qdka in application app:
NAME                 CONFIG      URL                                                             PORT
nodejs-qdka-8080     Present     http://nodejs-qdka-8080-app-myproject.192.168.42.180.nip.io     8080

$ odo url delete nodejs-qdka-8080
? Are you sure you want to delete the url nodejs-qdka-8080 Yes
URL removed from the config file
To delete URL from the OpenShift cluster, please run `odo push`

$ odo url list
Found the following URLs for component nodejs-qdka in application app:
NAME                 CONFIG     URL                                                             PORT
nodejs-qdka-8080     Absent     http://nodejs-qdka-8080-app-myproject.192.168.42.180.nip.io     8080
To delete URLS on the OpenShift Cluster, please use `odo push` 

$  odo url create --port 8000
 ✓  URL created for component: nodejs-qdka

To create URL on the OpenShift Cluster, please use `odo push` 

$ odo url list
Found the following URLs for component nodejs-qdka in application app:
NAME                 CONFIG      URL                                                             PORT
nodejs-qdka-8000     Present     <not created on cluster>                                        8000
nodejs-qdka-8080     Absent      http://nodejs-qdka-8080-app-myproject.192.168.42.180.nip.io     8080
To create/delete URLS on the OpenShift Cluster, please use `odo push` 


```

amitkrout commented:
@mohammedzee1000 Add/update test in the respective test files

surajnarwade commented on pkg/odo/cli/url/list.go:
```diff
@@ -75,29 +76,52 @@ func (o *URLListOptions) Run() (err error) {
 			return fmt.Errorf("no URLs found for component %v in application %v", o.Component(), o.Application)
 		} else {
-
+			var cm, dm bool
```
comment needed

mohammedzee1000 replied:
Added

surajnarwade commented on pkg/odo/cli/url/list.go:
```diff
@@ -75,29 +76,52 @@ func (o *URLListOptions) Run() (err error) {
 			tabWriterURL.Flush()
-			fmt.Println("\nUse `odo push` to create url on cluster")
+			if cm && dm {
+				printtemplates.PrintPushMessage("create/delete", "URLS")
```
s/URLS/URLs

mohammedzee1000 replied:
Addressed

surajnarwade commented:
travis failing, restarted the build

(project wait issue)

amitkrout commented:
```
[odo]  •  Waiting for project to come up  ...
```
/test e2e-core

amitkrout commented:
> @mohammedzee1000 Add/update test in the respective test files

Can you please add integration test scenario for the same

mohammedzee1000 commented:
sure i am looking at that right now

mohammedzee1000 commented:
Added

mohammedzee1000 commented:
/refresh

amitkrout commented on .travis.yml:
```diff
@@ -59,6 +59,7 @@ jobs:
         - travis_wait make test-source-e2e
         - travis_wait make test-json-format-output
         - travis_wait make test-odo-config
+        - travis_wait make test-odo-url-int
```
s/travis_wait make test-odo-url-int/travis_wait make test-cmd-url
Make more sense i guess

mohammedzee1000 replied:
I guess i can do that, but how would that make more sense than what is already there. Also, I am in favor of keeping test type in the name, making it explicit that integration test will be run and nothing else

amitkrout replied:
Right, that is way i suggested the name ```test-cmd-url```. The name it self says that you can not run e2e test over a command (e2e test runs on user specified scenario where you can run  multiple command/sub-command to achieve it). Integration test always runs on command/sub-command. 

So no need to over qualify the name ;)

amitkrout commented on Makefile:
```diff
@@ -153,6 +153,11 @@ test-odo-login-e2e:
 	go test -v github.com/openshift/odo/tests/integration --ginkgo.focus="odo config test" -ginkgo.slowSpecThreshold=$(SLOW_SPEC_THRESHOLD) -ginkgo.v -timeout $(TIMEOUT)
 
+# Run config tests
+.PHONY: test-odo-url-int
```
s/test-odo-url-int/test-cmd-url

amitkrout replied:
https://github.com/openshift/odo/pull/1743#discussion_r295797164

amitkrout commented on tests/integration/url_test.go:
```diff
@@ -0,0 +1,87 @@
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odoURLIntegration", func() {
```
> odoURLIntegration

+1

amitkrout commented on tests/integration/url_test.go:
```diff
@@ -0,0 +1,87 @@
+		helper.DeleteDir(context)
+	})
+
+	Context("Listing urls", func() {
```
Test spec looks too length due to lots of text validation. Can you please minimize those text assertion by keeping those text which are really matter for the spec and output. And more importantly keep all the url command scenario here those are present in other test file like ```generic_test.go```.

If possible add all possible url command and sub-command in this test file.

mohammedzee1000 replied:
Sure, I will see if i can shorten it.

mohammedzee1000 replied:
Yes, i guess it would be possible to do that, but `generic _test.go` contains a lot of tests that may as well reside in a specific file. I think it is ok to have that there and have URL specific tests here.

amitkrout replied:
> I think it is ok to have that there and have URL specific tests here.

You are right, I mean to say just check the UX if anything is missed in the URL specific test file. 

For example:
```
$ odo url -h
Expose component to the outside world. 

The URLs that are generated using this command, can be used to access the deployed components from outside the cluster.

Usage:
  odo url [flags]
  odo url [command]

Examples:
  # Create a URL for the current component with a specific port
  odo url create --port 8080
  
  # Create a URL with a specific name and port
  odo url create example --port 8080
  
  # Create a URL with a specific name by automatic detection of port (only for components which expose only one service port)
  odo url create example
  
  # Create a URL with a specific name and port for component frontend
  odo url create example --port 8080 --component frontend
  # Delete a URL to a component
  odo url delete myurl
  # List the available URLs
  odo url list

Available Commands:
  create      Create a URL for a component
  delete      Delete a URL
  list        List URLs

Flags:
  -h, --help   Help for url

Global Flags:
      --skip-connection-check   Skip cluster check
  -v, --v Level                 Log level for V logs. Level varies from 0 to 9 (default 0).
      --vmodule moduleSpec      Comma-separated list of pattern=N settings for file-filtered logging

Use "odo url [command] --help" for more information about a command.
```

amitkrout commented:
/refresh

mohammedzee1000 commented:
/refresh

mohammedzee1000 commented:
```
```

mohammedzee1000 commented:
An issue with a timeout during justbeforeeach
```
odoCmpE2e
/go/src/github.com/openshift/odo/tests/integration/cmp_test.go:13
  Test odo push with --source and --config flags
  /go/src/github.com/openshift/odo/tests/integration/component.go:113
    when using project flag(--project) and current directory
    /go/src/github.com/openshift/odo/tests/integration/component.go:124
      create local nodejs component and push source and code at once [JustBeforeEach]
      /go/src/github.com/openshift/odo/tests/integration/component.go:167

      Timed out after 600.000s.
      Expected process to exit.  It did not.

```
/retest

mohammedzee1000 commented:
Cluster failed
/retest

mohammedzee1000 commented:
That is an odd failure, unit tests dont fail locally
/test unit

mohammedzee1000 commented:
/refresh

mohammedzee1000 commented:
Failure seems to be unrelated
Possible flake
```
Summarizing 1 Failure:

[Fail] odoCmpSubE2e odo component updating [It] should be able to create a git component and update it from local to git 
/go/src/github.com/openshift/odo/tests/helper/helper_run.go:29

Ran 12 of 61 Specs in 844.566 seconds
FAIL! -- 11 Passed | 1 Failed | 0 Pending | 49 Skipped
--- FAIL: TestIntegration (844.57s)
FAIL
FAIL	github.com/openshift/odo/tests/integration	844.602s
make[1]: Leaving directory `/go/src/github.com/openshift/odo'
make[1]: *** [test-cmp-sub-e2e] Error 1
make: *** [openshiftci-presubmit-e2e] Error 2
2019/06/06 08:03:15 Container test in pod integration failed, exit code 2, reason Error
2019/06/06 08:08:15 Copied 149.20Mi of artifacts from integration to /logs/artifacts/integration
2019/06/06 08:08:18 Ran for 1h38m30s
2019/06/06 08:08:18 Submitted failure event to sentry (id=879a7f4da4f34bbaaff846403fc1e653)
error: could not run steps: step integration failed: template pod "integration" failed: the pod ci-op-20xsby8n/integration failed after 1h38m5s (failed containers: test): ContainerFailed one or more containers exited

Container test exited with code 2, reason Error
---
lications, which will be deleted
[odo] Application testing
[odo] This application has following components that will be deleted
[odo] component named cmp-git
[odo] No services / could not get services
[odo]  •  Deleting project aqixncmcan  ...
[odo]  ✓  Deleting project aqixncmcan
[odo] Deleted project : aqixncmcan

• Failure [131.540 seconds]
odoCmpSubE2e
/go/src/github.com/openshift/odo/tests/integration/cmp_sub_test.go:13
  odo component updating
  /go/src/github.com/openshift/odo/tests/integration/component.go:299
    should be able to create a git component and update it from local to git [It]
    /go/src/github.com/openshift/odo/tests/integration/component.go:310

    No future change is possible.  Bailing out early after 74.126s.
    Expected
        <int>: 1
    to match exit code:
        <int>: 0

    /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
```
/retest

mohammedzee1000 commented:
/retest

mohammedzee1000 commented:
> One last comment
> 
> Just i realized that we have already introduced parallel test run. Can you sync your test with the test template [1].
> 
> [1] https://github.com/openshift/odo/blob/master/tests/template/template_cleantest_test.go
> 
> **OR**
> 
> Refer [2] with Makefile change [3] which has already went in to master
> 
> [2] https://github.com/openshift/odo/blob/master/tests/integration/cmd_app_test.go
> [3] https://github.com/openshift/odo/blob/master/Makefile#L157

updated

mohammedzee1000 commented:
This is so weird. These tests pass locally.
```
$ make bin
go build -ldflags="-w -X github.com/openshift/odo/pkg/odo/cli/version.GITCOMMIT=2953b7ca" cmd/odo/odo.go
$ oc whoami
developer
$ make test-odo-url-int 
ginkgo -v -nodes=4 -focus="odoURLIntegration" slowSpecThreshold=120 -randomizeAllSpecs  tests/integration/ -timeout 1800s
Running Suite: Integration Suite
================================
Random Seed: 1561509768
Will run 84 specs

Running in parallel across 4 nodes

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
odoURLIntegration Listing urls 
  should list appropriate URLs and push message
  /home/mzee1000/workbench/work/golang/redhat/src/github.com/openshift/odo/tests/integration/url_test.go:59

Created dir: /tmp/078986391
Creating a new project: ocokhwssyp
Running odo with args: [project create ocokhwssyp -w -v4] 
[odo] I0626 06:12:51.020888    9514 preference.go:116] The configFile is /tmp/078986391/config.yaml
[odo] I0626 06:12:51.021016    9514 occlient.go:455] Trying to connect to server 192.168.42.6:8443
[odo] I0626 06:12:51.021954    9514 occlient.go:462] Server https://192.168.42.6:8443 is up
[odo] I0626 06:12:51.083177    9514 occlient.go:385] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0626 06:12:51.083229    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0626 06:12:51.083247    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0626 06:12:51.083370    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0626 06:12:51.083398    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0626 06:12:51.105235    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0626 06:12:51.105266    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0626 06:12:51.105275    9514 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [2562047h47m16.854775807s]
[odo]  ✓  Project 'ocokhwssyp' is ready for use
[odo] I0626 06:12:51.528997    9514 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  ✓  New project created and now using project : ocokhwssyp
Running oc with args: [project -q] 
[oc] ocokhwssyp
Switching to project : ocokhwssyp
Running oc with args: [project ocokhwssyp] 
[oc] Already on project "ocokhwssyp" on server "https://192.168.42.6:8443".
Running odo with args: [create nodejs --context /tmp/078986391 --project ocokhwssyp szkjfq --ref master --git https://github.com/openshift/nodejs-ex] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [push --context /tmp/078986391] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [2562047h47m16.854775807s]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [2562047h47m16.854775807s]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2562047h47m16.854775807s]
[odo]  •  Deploying component szkjfq  ...
[odo]  ✓  Deploying component szkjfq [2562047h47m16.854775807s]
[odo]  ✓  Creating component [2562047h47m16.854775807s]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [2562047h47m16.854775807s]
[odo] 
[odo] Pushing to component szkjfq of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/078986391] 
[odo]  ✗  no URLs found for component szkjfq in application app
Running odo with args: [url create htyvw --port 8080 --context /tmp/078986391] 
[odo]  ✓  URL htyvw created for component: szkjfq
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/078986391] 
[odo] Found the following URLs for component szkjfq in application app:
[odo] NAME      IN CONFIG     URL                          PORT
[odo] htyvw     Present       <not created on cluster>     8080
[odo] To create URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [push --context /tmp/078986391] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [2562047h47m16.854775807s]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [2562047h47m16.854775807s]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2562047h47m16.854775807s]
[odo]  ✓  Applying configuration [2562047h47m16.854775807s]
[odo] 
[odo] Applying URL changes
[odo]  ✓  URL htyvw: http://htyvw-app-ocokhwssyp.192.168.42.6.nip.io created
[odo] 
[odo] Pushing to component szkjfq of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/078986391] 
[odo] Found the following URLs for component szkjfq in application app:
[odo] NAME      IN CONFIG     URL                                                 PORT
[odo] htyvw     Present       http://htyvw-app-ocokhwssyp.192.168.42.6.nip.io     8080
Running odo with args: [url delete htyvw -f --context /tmp/078986391] 
[odo]  ✓  URL htyvw removed from the config file
[odo] To delete URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/078986391] 
[odo] Found the following URLs for component szkjfq in application app:
[odo] NAME      IN CONFIG     URL                                                 PORT
[odo] htyvw     Absent        http://htyvw-app-ocokhwssyp.192.168.42.6.nip.io     8080
[odo] To delete URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url create awids --port 8000 --context /tmp/078986391] 
[odo]  ✓  URL awids created for component: szkjfq
[odo] 
[odo] To create URL on the OpenShift Cluster, please use `odo push` 
Running odo with args: [url list --context /tmp/078986391] 
[odo] Found the following URLs for component szkjfq in application app:
[odo] NAME      IN CONFIG     URL                                                 PORT
[odo] awids     Present       <not created on cluster>                            8000
[odo] htyvw     Absent        http://htyvw-app-ocokhwssyp.192.168.42.6.nip.io     8080
[odo] To create/delete URLs on the OpenShift Cluster, please use `odo push` 
Running odo with args: [push --context /tmp/078986391] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [2562047h47m16.854775807s]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [2562047h47m16.854775807s]
[odo] 
[odo] Configuration changes
[odo]  •  Applying configuration  ...
[odo]  •  Triggering build from git  ...
[odo]  ✓  Triggering build from git [2562047h47m16.854775807s]
[odo]  •  Waiting for build to finish  ...
[odo]  ✓  Waiting for build to finish [2562047h47m16.854775807s]
[odo]  ✓  Applying configuration [2562047h47m16.854775807s]
[odo] 
[odo] Applying URL changes
[odo]  ✓  URL awids: http://awids-app-ocokhwssyp.192.168.42.6.nip.io created
[odo]  ✓  URL htyvw successfully deleted
[odo] 
[odo] Pushing to component szkjfq of type git
[odo]  ✓  Changes successfully pushed to component
Running odo with args: [url list --context /tmp/078986391] 
[odo] Found the following URLs for component szkjfq in application app:
[odo] NAME      IN CONFIG     URL                                                 PORT
[odo] awids     Present       http://awids-app-ocokhwssyp.192.168.42.6.nip.io     8000
Switching to project : ocokhwssyp
Running oc with args: [project ocokhwssyp] 
[oc] Already on project "ocokhwssyp" on server "https://192.168.42.6:8443".
Deleting project: ocokhwssyp
Running odo with args: [project delete ocokhwssyp -f] 
[odo]  ⚠  Unable to access Service Catalog instances, may not be enabled on cluster
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named szkjfq
[odo] This component has following urls that will be deleted with component
[odo] URL named awids with host awids-app-ocokhwssyp.192.168.42.6.nip.io having protocol http at port 8000
[odo] No services / could not get services
[odo]  •  Deleting project ocokhwssyp  ...
[odo]  ✓  Deleting project ocokhwssyp [2562047h47m16.854775807s]
[odo] Deleted project : ocokhwssyp
Deleating dir: /tmp/078986391


• [SLOW TEST:151.514 seconds]
odoURLIntegration
/home/mzee1000/workbench/work/golang/redhat/src/github.com/openshift/odo/tests/integration/url_test.go:13
  Listing urls
  /home/mzee1000/workbench/work/golang/redhat/src/github.com/openshift/odo/tests/integration/url_test.go:46
    should list appropriate URLs and push message
    /home/mzee1000/workbench/work/golang/redhat/src/github.com/openshift/odo/tests/integration/url_test.go:59
------------------------------

Ran 1 of 84 Specs in 151.541 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 83 Skipped


Ginkgo ran 1 suite in 2m33.739108081s
Test Suite Passed

```

mohammedzee1000 commented:
This is what is happening in CI

```
Creating a new project: tbhsphpcer
Running odo with args: [project create tbhsphpcer -w -v4] 
[odo] I0626 02:15:45.330802   27550 preference.go:116] The configFile is /tmp/artifacts/.odo/preference.yaml
[odo] I0626 02:15:45.330932   27550 occlient.go:455] Trying to connect to server api.ci-op-cbclhvzz-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443
[odo] I0626 02:15:45.347775   27550 occlient.go:462] Server https://api.ci-op-cbclhvzz-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443 is up
[odo] I0626 02:15:45.411993   27550 occlient.go:385] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0626 02:15:45.412035   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0626 02:15:45.412042   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0626 02:15:45.412103   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0626 02:15:45.412109   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0626 02:15:45.425862   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0626 02:15:45.425886   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0626 02:15:45.425891   27550 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [438ms]
[odo]  ✓  Project 'tbhsphpcer' is ready for use
[odo]  ✓  New project created and now using project : tbhsphpcer
Created dir: /tmp/094499100
Running odo with args: [component create nodejs nodejs --app nodeapp --project tbhsphpcer --context /tmp/094499100] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [361ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [storage create pv1 --path /mnt/pv1 --size 5Gi --context /tmp/094499100] 
[odo]  ✓  Added storage pv1 to nodejs
Running odo with args: [push --context /tmp/094499100] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [356ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [17ms]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Added storage pv1 to nodejs
[odo]  ✓  Initializing component
[odo]  ✓  Creating component [236ms]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [26899ns]
[odo] 
[odo] Pushing to component nodejs of type local
[odo]  •  Waiting for component to start  ...
[odo]  ✗  Waiting for component to start [4m]
[odo]  ✗  waited 4m0s but couldn't find running pod matching selector: 'deploymentconfig=nodejs-nodeapp'
Deleting project: tbhsphpcer
Running odo with args: [project delete tbhsphpcer -f] 
[odo] This project contains the following applications, which will be deleted
[odo] Application nodeapp
[odo]  ⚠  Unable to access Service Catalog instances, may not be enabled on cluster
[odo] This application has following components that will be deleted
[odo] component named nodejs
[odo] No services / could not get services
[odo]  •  Deleting project tbhsphpcer  ...
[odo]  ✓  Deleting project tbhsphpcer [6s]
[odo] Deleted project : tbhsphpcer
Deleating dir: /tmp/094499100
 • Failure [248.743 seconds]
odo storage command
/go/src/github.com/openshift/odo/tests/integration/cmd_storage_test.go:13
  when using storage command with default flag values
  /go/src/github.com/openshift/odo/tests/integration/cmd_storage_test.go:51
    should add a storage, list and delete it [It]
    /go/src/github.com/openshift/odo/tests/integration/cmd_storage_test.go:52
     No future change is possible.  Bailing out early after 240.896s.
      
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
     /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
------------------------------
 Summarizing 1 Failure:
 [Fail] odo storage command when using storage command with default flag values [It] should add a storage, list and delete it 
/go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
 Ran 5 of 84 Specs in 248.771 seconds
FAIL! -- 4 Passed | 1 Failed | 0 Pending | 79 Skipped
```

mohammedzee1000 commented:
/refresh

amitkrout commented:
/test integration

mohammedzee1000 commented:
/retest

amitkrout commented:
Test wise looks good to me. If any improvement is needed in integration test then that will be fixed in a separate pr

/lgtm

amitkrout commented:
/test integration

amitkrout commented:
/refresh
